x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedJob.java

```diff
@@ -223,10 +224,23 @@ private void checkForMissingDataIfNecessary() {
                     && annotation.getEndTimestamp().equals(lastDataCheckAnnotationWithId.v2().getEndTimestamp())) {
                     return;
                 }
-
-                // Creating a warning in addition to updating/creating our annotation. This allows the issue to be plainly visible
-                // in the job list page.
-                auditor.warning(jobId, msg);
+                if (lastDataCheckAnnotationWithId != null) {
+                    long bucketAfterPreviousAnnotation = lastDataCheckAnnotationWithId.v2().getEndTimestamp().getTime()
```

Comment:
Is it bucket start? bucket end? Could you rename this variable to convey its semantics better?

Reply:
I agree this is a bit confusing. Is it safe to assume an annotation's end time is the start of the next bucket? If yes, could we compare the previous annotations end time with the current one's start time?

Reply:
> with the current one's start time?

No, we cannot. In the case of consecutive buckets missing data, the start time stays at the first buckets start time.

What I could do is check the floor of `lastBucket` vs the endTime of `lastDataCheckAnnotationWithId`. That would indicate if this latest bucket overlaps with the endtime of the previous annotation.